linux/LinuxProcessList.c

```diff
@@ -1535,11 +1636,12 @@ static bool LinuxProcessList_recurseProcTree(LinuxProcessList* this, openat_arg_
          continue;
       }
 
-      bool scanMainThread = !hideUserlandThreads && !Process_isKernelThread(proc) && !parent;
+      const bool scanMainThread = !hideUserlandThreads && !Process_isKernelThread(proc) && !mainTask;
```

Comment:
What's the purpose of the `const` qualifier here?

Reply:
@BenBE There is no benefit to tag a local variable as a `const`, even for the purpose of compiler optimization. And while I agree the tag prevents the programmer from accidentally reassign the value later in the code, the tag is no longer needed when the code is submitted for version control. You know, any lines added for reassigning a value of a variable would have been intended by the programmer, or otherwise the patch is faulty already and should not be merged to `main`.

Reply:
I kinda disagree regarding the "does not belong in the patch": It's not just about the state of the code when it was written, but also going forward. And needing to consciously remove the const when you want to write to this variable in some later code change is better, as it communicates intent.

Reply:
It was added as a mental shortcut: when looking over the code (maybe after some time has passed) it directly states the variable is not going to change, and one can skip checking all possible branches in between the assignment and any of the usages for potential modifications.
So it's more of a maintainability thing than a (compiler) optimization.